### ISSUE TITLE
Peacock writes extra settings

### ISSUE THREAD

**Issue by johnpapa** [johnpapa/vscode-peacock#229]
Versions:
- Mac: 10.14.5
- vscode: 1.36.1
- Peacock: 2.5.2

User @kbd notices Peacock saves 👇

```
          "editor.selectionBackground": "#888",
          "editor.selectionHighlightBackground": "#555",
          "editor.findMatchHighlightBackground": "#2cdd1c65",
          "editorIndentGuide.activeBackground": "#888"
```

Peacock would only save those if they are already in your settings hierarchy defaults, global user settings, workspace settings , etc.

This is here because if you have settings, Peacock doesn't want to overwrite those. Perhaps, and I am guessing because I dont have a complete picture here, those settings for color customizations exist somewhere else in the hierarchy and Peacock is grabbing those and re-writing them. if so, that is a bug. 

Can you clarify where those exist (before Peacock) ?

**Comment by kbd** [comment]
> Can you clarify where those exist (before Peacock) ?

Those look like they're coming from [my global color customizations](https://github.com/kbd/setup/blob/efb57ef769c3ddbf397e3d8e0ca087f250ce08a0/HOME/Library/Application%20Support/Code/User/settings.json#L73). But only 3.0 (not 2.5.2) -- even with "surprise me on startup" set to false -- creates a `.vscode/settings.json` if one doesn't exist.

**Comment by johnpapa** [comment]
Thanks.

I have confirmed this is a bug. Reproduced like this:

*Global user settings.json*
```json
    "workbench.colorCustomizations": {
        "panel.background": "#212121"
    }
```

*Workspace settings.json*
```json
    "workbench.colorCustomizations": {
        "breadcrumb.foreground": "#ff0000"
    }
```

When Peacock resets the colors, it results in this

*Workspace settings.json*
```json
    "workbench.colorCustomizations": {
        "panel.background": "#212121",
        "breadcrumb.foreground": "#ff0000"
    }
```

Why?

Because when it resets the colors it grabs all workbench colorcustomizations. VS Code's API gets a merged set of these from all sources. Peacock is then removing the Peacock ones, and then writing the rest to the workspace settings.json. This is not the desired behavior.


Solutions

There are a few options.

- Perhaps the inspect API could be used to determine where each setting comes from and we only write ones from the workspace settings. I have had limited success with inspect on 4 level's of objects tho "workbench.colorCustomizations.panel.background". I may have to ping the VS Code team on this one.
- I could wipe  all customizations out in the workspace. But that would get rid of any extra ones the user may have created (like the breadcrumb above). So this is not desirable.
- The goal is to remove the Peacock colors. Another option is to loop through all of them and delete them one by one, writing to the workspace 1 at a time. This works, but writes 10+ times. This may be the most accurate tho.
